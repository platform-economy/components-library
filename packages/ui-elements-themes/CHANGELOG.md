# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
