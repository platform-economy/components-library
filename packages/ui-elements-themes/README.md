# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
