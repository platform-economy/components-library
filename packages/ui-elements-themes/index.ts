export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
