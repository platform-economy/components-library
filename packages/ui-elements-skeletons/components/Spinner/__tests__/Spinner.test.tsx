import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { withTestTheme } from '@relayr/ui-elements-themes';
import 'jest-styled-components';

// Component
import { Spinner, sizeCounter, borderSizeCounter } from '../index';

describe('ui-elements-skeletons', () => {
  describe('Components', () => {
    describe('Spinner', () => {

      it('should render with default state', () => {
        const tree = TestRenderer
          .create(withTestTheme(<Spinner />))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
      it('should render with size prop as number', () => {
        const tree = TestRenderer
          .create(withTestTheme(<Spinner size={300}/>))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
      it('should render with size prop as text', () => {
        const tree = TestRenderer
          .create(withTestTheme(<Spinner size="100px"/>))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
      it('should count size properly', () => {
        expect(sizeCounter(10)).toBe('10px');
        expect(sizeCounter('100px')).toBe('100px');
        expect(sizeCounter('100em')).toBe('100em');
        expect(sizeCounter(undefined)).toBe('1em');
      });
      it('should count border size properly', () => {
        expect(borderSizeCounter(80)).toBe('10px');
        expect(borderSizeCounter('80px')).toBe('10px');
        expect(borderSizeCounter('80em')).toBe('10em');
        expect(borderSizeCounter(undefined)).toBe('.2em');
      });
    });
  });
});
