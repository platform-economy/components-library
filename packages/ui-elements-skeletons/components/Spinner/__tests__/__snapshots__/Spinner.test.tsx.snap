// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-skeletons Components Spinner should render with default state 1`] = `
.c0 {
  height: 1em;
  width: 1em;
  border-radius: 50%;
  border: .2em solid #text_inverted;
  border-top: .2em solid #active;
  -webkit-animation: hmHMaM 1s infinite linear;
  animation: hmHMaM 1s infinite linear;
  box-sizing: border-box;
}

<div
  className="c0"
/>
`;

exports[`ui-elements-skeletons Components Spinner should render with size prop as number 1`] = `
.c0 {
  height: 300px;
  width: 300px;
  border-radius: 50%;
  border: 37.5px solid #text_inverted;
  border-top: 37.5px solid #active;
  -webkit-animation: hmHMaM 1s infinite linear;
  animation: hmHMaM 1s infinite linear;
  box-sizing: border-box;
}

<div
  className="c0"
  size={300}
/>
`;

exports[`ui-elements-skeletons Components Spinner should render with size prop as text 1`] = `
.c0 {
  height: 100px;
  width: 100px;
  border-radius: 50%;
  border: 12.5px solid #text_inverted;
  border-top: 12.5px solid #active;
  -webkit-animation: hmHMaM 1s infinite linear;
  animation: hmHMaM 1s infinite linear;
  box-sizing: border-box;
}

<div
  className="c0"
  size="100px"
/>
`;
